### PR TITLE
Better config for admin RSS links

### PR DESCRIPTION
**Pull request by sreichel**
### Description (*)

Description here: #2760

### Related Pull Requests

1. See OpenMage/magento-lts#2760
2. See OpenMage/magento-lts#3520
3. See OpenMage/magento-lts#3997

### CODE REVIEW

**Review by ADDISON74 — APPROVED** [review]

**Review by kiatng — APPROVED** [review]

File: app/code/core/Mage/Adminhtml/Block/Sales/Order/Grid.php [hunk 1]
```diff
@@ -141,11 +141,15 @@ protected function _prepareColumns()
...
-            Mage::helper('rss')->isRssEnabled() &&
-            Mage::getStoreConfigFlag('rss/order/new')
-        ) {
-            $this->addRssList('rss/order/new', Mage::helper('sales')->__('New Order RSS'));
+        if (Mage::helper('sales')->isModuleOutputEnabled('Mage_Rss')) {
+            $filterString = $this->getParam($this->getVarNameFilter());
+            $filter = Mage::helper('adminhtml')->prepareFilterString($filterString);
+            $storeId = array_key_exists('store_id', $filter) ? $filter['store_id'] : null;
+
+            if (Mage::helper('rss')->isRssAdminOrderNewEnabled($storeId)) {
```
Comment (fballiano) [inline app/code/core/Mage/Adminhtml/Block/Sales/Order/Grid.php:149]: if ($storeId && Mage::helper('rss')->isRssAdminOrderNewEnabled($storeId)) {

File: app/code/core/Mage/Adminhtml/Block/Sales/Order/Grid.php [hunk 2]
```diff
@@ -141,11 +141,15 @@ protected function _prepareColumns()
             );
         }
 
-        if (Mage::helper('catalog')->isModuleEnabled('Mage_Rss') &&
-            Mage::helper('rss')->isRssEnabled() &&
-            Mage::getStoreConfigFlag('rss/order/new')
-        ) {
-            $this->addRssList('rss/order/new', Mage::helper('sales')->__('New Order RSS'));
+        if (Mage::helper('sales')->isModuleOutputEnabled('Mage_Rss')) {
+            $filterString = $this->getParam($this->getVarNameFilter());
```
[the remark this answers is not in the document]
Comment (kiatng) [reply app/code/core/Mage/Adminhtml/Block/Sales/Order/Grid.php:145]: @sreichel Can you fix:
```
    [type] => 8192:E_DEPRECATED
    [message] => base64_decode(): Passing null to parameter #1 ($string) of type string is deprecated
    [file] => app\code\core\Mage\Adminhtml\Helper\Data.php
    [line] => 108
```

File: app/code/core/Mage/Adminhtml/Block/Sales/Order/Grid.php (hunk 1 above)
Comment (sreichel) [reply app/code/core/Mage/Adminhtml/Block/Sales/Order/Grid.php:149]: $storeId  is used one line later.

```
            if (Mage::helper('rss')->isRssAdminOrderNewEnabled($storeId)) {
                $slug = $storeId ? '/store/' . $storeId : '';
                $this->addRssList('rss/order/new' . $slug, Mage::helper('sales')->__('New Order RSS'));
            }
```

File: app/code/core/Mage/Adminhtml/Block/Sales/Order/Grid.php (hunk 2 above)
[the remark this answers is not in the document]
Comment (sreichel) [reply app/code/core/Mage/Adminhtml/Block/Sales/Order/Grid.php:145]: PR comming. How to reproduce?

[the remark this answers is not in the document]
Comment (fballiano) [reply app/code/core/Mage/Adminhtml/Block/Sales/Order/Grid.php:145]: go to the sales->order backend page...

**Pull request merged**